### PR TITLE
add _Fillvalue

### PR DESCRIPTION
**Pull request by hailingz**
This PR was intended to address https://github.com/JCSDA/ioda-converters/issues/233

Changes include:
1)  add _FillValue attributes in needed variables in the GNSSRO converter. 
2) add many other attributes for variables, e.g., longname, units, valid_range, for a more clear description.
  **  @fcvdb  @huishao-r if you have time please review those attributes (Line 315-370, https://github.com/JCSDA/ioda-converters/blob/feature/gnssro_missing_value/src/gnssro/gnssro_bufr2ioda.f90). Thanks.

This merge of this PR will close https://github.com/JCSDA/ioda-converters/issues/233

### CODE REVIEW

File: src/gnssro/gnssro_bufr2ioda.f90 [hunk 1]
```diff
@@ -311,6 +311,7 @@ program gnssro_bufr2ioda
 call check( nf90_def_dim(ncid, 'nvars', nvars,   nvars_dimid) )
 call check( nf90_def_dim(ncid, 'ndatetime', ndatetime,   ndatetime_dimid) )
 call check( nf90_put_att(ncid, NF90_GLOBAL, 'date_time', anatime_i) )
+call check( nf90_put_att(ncid, NF90_GLOBAL, '_FillValue', missingvalue) )
```
Comment (srherbener) [inline src/gnssro/gnssro_bufr2ioda.f90]: The _FillValue attribute needs to be added to each variable that contains missing values. It won't have an effect as a global attribute.

One quick way to test if you've got it right is to run ncdump on the output file. If a variable with value equal to missingvalue shows up with a "-" in the output of ncdump, then the _FillValue attribute is working as intended. If instead the value shows up as what you would expect for missingvalue (-9.9e15) then it is not being recognized as the fill value.

Comment (hailingz) [reply src/gnssro/gnssro_bufr2ioda.f90]: got it! Thank you very much!

Comment (jjguerrette) [reply src/gnssro/gnssro_bufr2ioda.f90]: Just to clarify on @srherbener's description.  The locations equal to _FillValue show up as "_", an underscore, not a hyphen.

**Comment by jjguerrette** [comment]
I added @svahl991 and @junmeiban, because one of them will likely do the conversions for our group and can test the updated code.

File: src/gnssro/gnssro_bufr2ioda.f90 (hunk 1 above)
Comment (hailingz) [reply src/gnssro/gnssro_bufr2ioda.f90]: I have added _FillValue to refractivity as well as many other attributes. Please review and try @srherbener @jjguerrette @svahl991 
Now I see the "_"s in ncdump.

**Comment by junmeiban** [comment]
Thanks @hailingz . I converted bufr format obs to IODA obs, the FillValue now shows up as "_". 
I also use the converted nc file to run DA experiment, it can run successfully.

**Comment by srherbener** [comment]
@hailing, when I build and run tests on my iMac, I see the following failures:

```
The following tests FAILED:
	 18 - test_iodaconv_gnssro_bufr_conv (Failed)
```

In the LastTest.log file:

```
DIFFER : VARIABLE "time@MetaData" IS MISSING ATTRIBUTE WITH NAME "longname" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : VARIABLE "time@MetaData" IS MISSING ATTRIBUTE WITH NAME "units" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : NUMBER OF ATTRIBUTES : VARIABLE : time@MetaData : 2 <> 0
DIFFER : VARIABLE "record_number@MetaData" IS MISSING ATTRIBUTE WITH NAME "longname" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : NUMBER OF ATTRIBUTES : VARIABLE : record_number@MetaData : 1 <> 0
DIFFER : VARIABLE "gnss_sat_class@MetaData" IS MISSING ATTRIBUTE WITH NAME "longname" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : NUMBER OF ATTRIBUTES : VARIABLE : gnss_sat_class@MetaData : 1 <> 0
DIFFER : VARIABLE "reference_sat_id@MetaData" IS MISSING ATTRIBUTE WITH NAME "longname" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : NUMBER OF ATTRIBUTES : VARIABLE : reference_sat_id@MetaData : 1 <> 0
DIFFER : VARIABLE "occulting_sat_id@MetaData" IS MISSING ATTRIBUTE WITH NAME "longname" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : NUMBER OF ATTRIBUTES : VARIABLE : occulting_sat_id@MetaData : 1 <> 0
DIFFER : VARIABLE "ascending_flag@MetaData" IS MISSING ATTRIBUTE WITH NAME "longname" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : VARIABLE "ascending_flag@MetaData" IS MISSING ATTRIBUTE WITH NAME "valid_range" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : NUMBER OF ATTRIBUTES : VARIABLE : ascending_flag@MetaData : 2 <> 0
DIFFER : VARIABLE "refractivity@ObsValue" IS MISSING ATTRIBUTE WITH NAME "longname" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : VARIABLE "refractivity@ObsValue" IS MISSING ATTRIBUTE WITH NAME "_FillValue" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : VARIABLE "refractivity@ObsValue" IS MISSING ATTRIBUTE WITH NAME "units" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : VARIABLE "refractivity@ObsValue" IS MISSING ATTRIBUTE WITH NAME "valid_range" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : NUMBER OF ATTRIBUTES : VARIABLE : refractivity@ObsValue : 4 <> 0
DIFFER : VARIABLE "refractivity@ObsError" IS MISSING ATTRIBUTE WITH NAME "longname" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : VARIABLE "refractivity@ObsError" IS MISSING ATTRIBUTE WITH NAME "_FillValue" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : VARIABLE "refractivity@ObsError" IS MISSING ATTRIBUTE WITH NAME "units" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : VARIABLE "refractivity@ObsError" IS MISSING ATTRIBUTE WITH NAME "valid_range" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : NUMBER OF ATTRIBUTES : VARIABLE : refractivity@ObsError : 4 <> 0
DIFFER : VARIABLE "altitude@MetaData" IS MISSING ATTRIBUTE WITH NAME "longname" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : VARIABLE "altitude@MetaData" IS MISSING ATTRIBUTE WITH NAME "units" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
DIFFER : NUMBER OF ATTRIBUTES : VARIABLE : altitude@MetaData : 2 <> 0
DIFFER : VARIABLE "bending_angle@ObsValue" IS MISSING ATTRIBUTE WITH NAME "longname" IN FILE "testoutput/gnssro_kompsat5_2018041500.nc4"
.
.
.
```

which is the output from nccmp. Looks like nccmp is flagging the differences being the new variable attributes that you added.  It appears that the test reference file needs to be updated.

I don't understand why the AWS tests are passing. @mer-a-o, is it possible that the AWS tests are running on the develop branch? In the log file for the AWS GNU test, I see this:

```
[Container] 2019/11/22 12:33:13 Running command echo "CODEBUILD_GIT_BRANCH=${CODEBUILD_GIT_BRANCH}" 
CODEBUILD_GIT_BRANCH=feature/gnssro_missing_value 
```

also this later on:

```
[Container] 2019/11/22 12:33:15 Running command git clone https://github.com/jcsda/ioda-converters 
```

but I don't see where it switches the branch to `feature/gnssro_missing_value`.

I might be wrong in thinking that the develop branch is being tested, but thought it would be good to double check. Thanks!

**Comment by hailingz** [comment]
@srherbener 
Thank you for your review. I have pushed the new testout file and please test again!

**Comment by srherbener** [comment]
This is all working now so I'm going to merge. The Travis-CI test failures are a configuration problem and there is a fix in another PR.

**Pull request merged**